File: leetcode2/0120. Triangle.py
class Solution:
    def minimumTotal(self, triangle: 'List[List[int]]') -> 'int':
        row=1
        while row<len(triangle):
            triangle[row][0]+=triangle[row-1][0]
            triangle[row][-1]+=triangle[row-1][-1]
            for column in range(1,len(triangle[row])-1):
                triangle[row][column]+=min(triangle[row-1][column-1],triangle[row-1][column])
            row+=1
        return min(triangle[row-1])

File: archives/C#/0120. Triangle.cs
public class Solution {
    public int MinimumTotal(IList<IList<int>> triangle) {
        for(int i=1;i<triangle.Count();i++){
            int len=triangle[i].Count();
            triangle[i][0]+=triangle[i-1][0];
            triangle[i][len-1]+=triangle[i-1][len-2];
            for(int j=1;j<len-1;j++){
                triangle[i][j]+=Math.Min(triangle[i-1][j-1],triangle[i-1][j]);
            }
        }
        return triangle[triangle.Count-1].Min();
    }
}
